common/grammar-parser.cpp

```diff
@@ -174,11 +174,11 @@ namespace grammar_parser {
             //            Sstar  ::= Scopy Sstar |
 
             uint32_t content_rule_id = 0;
-            if (last_sym_start == out_elements.size() - 1) {
+            if (last_sym_start >= 0 && last_sym_start == out_elements.size() - 1 && out_elements[last_sym_start].type == LLAMA_GRETYPE_RULE_REF) {
                 // The repeated content is already a rule ref, no need to copy it
                 content_rule_id = out_elements[last_sym_start].value;
             } else {
-                content_rule_id = generate_symbol_id(state, rule_name);
+                content_rule_id = generate_symbol_id(state, rule_name + "_copy");
```

Comment:
+1 -- One nice thing about using "_copy" for the name here is that underscores are not valid for rule names, so you're guaranteed that you won't get collisions with this. I like it!

Reply:
Note that the generate symbol id helper takes care of avoiding collisions anyway